scripts/pylib/twister/twisterlib/testplan.py

```diff
@@ -656,8 +635,22 @@ def apply_filters(self, **kwargs):
                     if not set(ts.modules).issubset(set(self.modules)):
                         instance.add_filter(f"one or more required modules not available: {','.join(ts.modules)}", Filters.TESTSUITE)
 
-                if runnable and not instance.run:
-                    instance.add_filter("Not runnable on device", Filters.PLATFORM)
+                # instance.run tells twister if a test can be build and executed (if true) or only build (if false)
+                instance.run, verdict = instance.check_runnable(self.options.enable_slow, runnable)
+
+                harness_supported, verdict_harness = instance.is_harness_supported()
+                if not harness_supported:
+                    instance.run = False
+                    verdict = verdict_harness
+
+                fixture_runnable, verdict_fixture = instance.is_fixture_available(plat.name, self.options.fixture, self.hwm.duts)
+                if not fixture_runnable:
+                    instance.run = False
+                    verdict = verdict_fixture
+
+                if runnable:
+                    if not instance.run:
+                        instance.add_filter(f"Not runnable on device. {verdict}", Filters.PLATFORM)
```

Comment:
Minor nitpick: probably we can rephrase is to:
```
"Not runnable on device because: {verdict}"
```

Currently it may look like two separate issues are reported, so it may me a bit confusing:
```
Not runnable on device. Harness led is not supported.
Not runnable on device. nsimdrv not found.
```

Instead it would be:
```
Not runnable on device because: Harness led is not supported.
Not runnable on device because: nsimdrv not found.
```

Feel free to ignore :)

Reply:
Good point. Changed.

Reply:
Hmm, probably you have forgotten to add it to PR - at least I don't see any changes the code related to this log message printing / generation.